lib/internal/mime.js

```diff
@@ -0,0 +1,279 @@
+      test(NotHTTPQuotedStringCodePoint, parameterValue) === false &&
+      parameterName in parameters === false) {
+      parameters[parameterName] = parameterValue;
+    }
+  }
+  target.parameters = ObjectEntries(parameters);
+  return target;
+}
+
+const test = (pattern, value) => {
+  pattern.lastIndex = 0;
+  const ret = pattern.test(value);
+  // console.dir({
+  //   pattern,
+  //   value,
+  //   _: [...value].map((c) => c.codePointAt(0)),
+  //   ret,
+  //   _1: (pattern.lastIndex = 0, NotHTTPTokenCodePoint.exec(value))
+  // });
+  return ret;
+};
+const exec = (pattern, value) => {
+  pattern.lastIndex = 0;
+  const ret = pattern.exec(value);
+  // console.dir({
+  //   pattern,
+  //   value,
+  //   _: [...value].map((c) => c.codePointAt(0)),
+  //   ret,
+  //   _1: (pattern.lastIndex = 0, NotHTTPTokenCodePoint.exec(value))
+  // });
+  return ret;
+};
+
+const encode = (value) => {
+  if (value.length === 0) return '""';
+  NotHTTPTokenCodePoint.lastIndex = 0;
+  const encode = test(NotHTTPTokenCodePoint, value);
+  if (!encode) return value;
+  const escaped = StringPrototypeReplace(value, /["\\]/g, '\\$&');
+  const ret = `"${escaped}"`;
+  return ret;
+};
+
+const MIMEStringify = (type, subtype, parameters) => {
+  let ret = `${type}/${subtype}`;
+  const entries = MIMEParamsData(parameters).entries();
+  let keyValuePair, done;
+  for (;;) {
+    ({ value: keyValuePair, done } =
+        MapIteratorPrototypeNext(entries));
+    if (done) break;
+    const [key, value] = keyValuePair;
+    const encoded = encode(value);
+    ret += `;${key}=${encoded}`;
+  }
+  return ret;
+};
+
+class MIMEParams {
+  #data;
+  constructor() {
+    this.#data = new Map();
+  }
+
+  delete(name) {
+    this.#data.delete(name);
+  }
+
+  get(name) {
+    const data = this.#data;
+    if (data.has(name)) {
+      return data.get(name);
+    }
+    return null;
+  }
+
+  has(name) {
+    return this.#data.has(name);
+  }
+
+  set(name, value) {
+    const data = this.#data;
+    NotHTTPTokenCodePoint.lastIndex = 0;
+    name = `${name}`;
+    value = `${value}`;
+    const invalidName = test(NotHTTPTokenCodePoint, name);
+    if (name.length === 0 || invalidName) {
+      throw new Error('Invalid MIME parameter name');
+    }
+    NotHTTPQuotedStringCodePoint.lastIndex = 0;
+    const invalidValue = test(NotHTTPQuotedStringCodePoint, value);
+    if (invalidValue) {
+      throw new Error('Invalid MIME parameter value');
+    }
+    data.set(name, value);
+  }
+
+  *entries() {
+    return yield* this.#data.entries();
+  }
+
+  *keys() {
+    return yield* this.#data.keys();
+  }
+
+  *values() {
+    return yield* this.#data.values();
+  }
+
+  *[SymbolIterator]() {
+    return yield* this.#data.entries();
+  }
+
+  static _data(o) {
+    return o.#data;
+  }
+}
+
+const MIMEParamsData = MIMEParams._data;
+delete MIMEParams._data;
+
+class MIME {
+  #type;
+  #subtype;
+  #parameters;
```

Comment:
What is the minimum Node version that ships private fields?

Reply:
v12.0.0